api/campaigns/class-maybe-show-campaign.php

```diff
@@ -73,27 +73,42 @@ function ( $post_data ) use ( $visit ) {
 			);
 		}
 
-		$page_referer_url = isset( $_REQUEST['ref'] ) ? $_REQUEST['ref'] : ''; // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
-
+		$referer_url                   = filter_input( INPUT_SERVER, 'HTTP_REFERER', FILTER_SANITIZE_STRING );
+		$page_referer_url              = isset( $_REQUEST['ref'] ) ? $_REQUEST['ref'] : ''; // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$all_segments                  = isset( $settings->all_segments ) ? $settings->all_segments : [];
+		$best_priority_segment         = $this->get_best_priority_segment( $all_segments, $client_id, $referer_url, $page_referer_url, $view_as_spec );
```

Comment:
Maybe we could skip calling `get_best_priority_segment` if previewing, and simply assign 
```
$best_priority_segment = view_as_spec['segment'];
```
?

Reply:
Good idea—I had some logic to assign the `view_as_spec['segment']` if it exists inside the `get_best_priority_segment` method, but it can be simplified further. How about immediately returning the `view_as_spec['segment']` value from that method if it exists, like in 5b51622?